### PR TITLE
CORE-9185 Replace JobRequest.Job with a "JobDetails" struct.

### PR DESCRIPTION
**Pull request by psarando**
This PR will replace the `JobRequest.Job` field with a `JobDetails` struct, containing only the fields of the job JSON required for messages, refactor `UpdateMessage` with the `JobDetails` struct, and reimplement the `NewLaunchRequest` function and its test.

### CODE REVIEW

**Review by johnworth — APPROVED** [review]
Nothing obviously wrong leaps out at me. :shipit:

File: amqp.go
Comment (ianmcorvidae) [inline amqp.go]: It might be worth preserving the json annotations that are on this in the model library (same with below lines). e.g. this one is the `uuid` key.

File: amqp.go
Comment (ianmcorvidae) [inline amqp.go]: I'm not sure what's supposed to go in here. The Job struct has UserID and UserGroups, but no User. It also has Submitter, which goes into the `username` json field.

File: amqp.go
Comment (ianmcorvidae) [inline amqp.go:255]: Was this just removed since v4? I'm mildly confused that it's a complete re-adding.

**Review by ianmcorvidae — CHANGES_REQUESTED** [review]
Couple of small things, but the basic idea seems good.

File: amqp.go
Comment (psarando) [reply amqp.go]: I'm not sure either. These fields were copied out of the `UpdateMessage` struct added in commit 8a8745f63506af9a90024656a3ea30089c79f309.

Since nothing appears to be using it yet, I think we can replace it with those other user fields. In fact, I just noticed that `UserGroups` is required by `FormatUserGroups`, which is used in a template in `condor-launcher`, so I think I'll need to hunt down anymore fields required by templates before merging this PR.

Comment (ianmcorvidae) [reply amqp.go]: Cool, sounds good to me. I see you thumbs-up'd the note about the json keys, but thinking about it more that's definitely necessary, so definitely copy those for everything you bring over.

Comment (ianmcorvidae) [reply amqp.go]: As far as my update message stuff, I only need the invocation ID, so as long as you keep that around I'm happy, heh

File: amqp.go
Comment (psarando) [reply amqp.go:255]: It was removed in commit 8a8745f63506af9a90024656a3ea30089c79f309, but v2 was branched off that commit's parent, then v3 was branched off v2, and v4 was branched off v3 (because all those changes still required the Job model in the messages).

File: amqp.go
Comment (psarando) [reply amqp.go]: Oh wait a minute. Part of CORE-9185 is to update condor-launcher to lookup job JSON from the database instead of AMQP messages. So I don't think we need any fields except `InvocationID` for now.

If we need to add more fields later, it can be added in a minor version bump.

Comment (ianmcorvidae) [reply amqp.go]: Oh, yeah, true, as long as it's looking up the JSON by invocation ID (or something derivable from it) then that should be all that's needed. Let's still keep the JobDetails struct type I think though, since it'll keep the old messages complaint (just with more information).

Comment (psarando) [reply amqp.go]: Sounds good to me. I just noticed that `condor-launcher` can also add `CondorID` to job launch update messages (which it's not doing yet). So I'll keep that one too.

**Review by ianmcorvidae — APPROVED** [review]

**Comment by ianmcorvidae** [comment]
LGTM after changes! :+1:

**Comment by psarando** [comment]
Thanks for the review 👍

**Pull request merged**
